sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -238,7 +240,8 @@ case class AlterTableAddColumnsCommand(
         // TextFileFormat only default to one column "value"
         // Hive type is already considered as hive serde table, so the logic will not
         // come in here.
-        case _: JsonDataSourceV2 | _: CSVDataSourceV2 | _: ParquetFileFormat | _: OrcDataSourceV2 =>
+        case _: CSVFileFormat | _: JsonFileFormat | _: ParquetFileFormat =>
+        case _: JsonDataSourceV2 | _: CSVDataSourceV2 | _: OrcDataSourceV2 =>
```

Comment:
These V2 data sources also support ADD COLUMNs? Do we have the test cases?

Reply:
No, V2 doesn't support ADD COLUMN. If it requires catalog support, Spark will fall back V2 to V1. 

Currently the result of `DataSource.lookupDataSource` for "csv"/"json"/"orc" will always be `CSVDataSourceV2`/`JsonDataSourceV2`/`OrcDataSourceV2`. So we need to match them here.